logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/mlsmigration/MLSMigrator.kt

```diff
@@ -70,15 +81,14 @@ class MLSMigratorImpl(
             }
     }
 
-    private suspend fun updateProtocolToMixed(conversationId: ConversationId) =
-        wrapApiRequest { conversationApi.updateProtocol(conversationId.toApi(), ConvProtocol.MIXED) }
-            .flatMap {
-                if (it is UpdateConversationProtocolResponse.ProtocolUpdated) {
-                    conversationRepository.fetchConversation(conversationId)
-                } else {
-                    Either.Right(Unit)
-                }
+    private suspend fun finalise(conversationId: ConversationId): Either<CoreFailure, Unit> {
```

Comment:
do we have recovery in case of failure?

Reply:
We try again in 24h or someone else in the group does it.